easybuild/easyblocks/o/openfoam.py

```diff
@@ -92,11 +92,8 @@ def __init__(self, *args, **kwargs):
             self.openfoamdir = '-'.join([self.name, '-'.join(self.version.split('-')[:2])])
         self.log.debug("openfoamdir: %s" % self.openfoamdir)
 
-        # Set build type to requested value
-        if self.toolchain.options['debug']:
-            self.build_type = 'Debug'
-        else:
-            self.build_type = 'Opt'
+        # Always set build type to Opt, let easybuild decide CFLAGS and CXXFLAGS
```

Comment:
Hmm, I don't see a need for this, especially in EasyBuild 5.0 where we have a global setting to determine the value of the `debug` toolchain option (which is disabled by default).

What's the motivation for this exactly?

Reply:
```
        if self.toolchain.options['debug']:
            self.build_type = 'Debug'
```
is just wrong logic. `self.toolchain.options['debug']` means "enable_debug_symbols" 
```
        if self.toolchain.options['noopt']:
            self.build_type = 'Debug'
```
would be appropriate.